src/main/java/marquez/api/JobResource.java

```diff
@@ -264,9 +265,11 @@ void throwIfExists(
     }
   }
 
-  void throwIfNotExists(@NonNull RunId runId) throws MarquezServiceException {
-    if (!jobService.runExists(runId)) {
-      throw new RunNotFoundException(runId);
+  void throwIfNotExists(@Nullable RunId runId) throws MarquezServiceException {
+    if (runId != null) {
```

Comment:
this seems unintuitive to me. If we want this method to throw an error when the runid is not found, we should probably expect to call this only in a context where the run exists. I would call `if (!null)` before calling throwIfNotExists.

Reply:
Good point. We can do something more like:

```
jobMeta.getRunId().ifPresent(this::throwIfNotExists);
```